extensions/http/src/main/java/org/mule/extension/http/api/request/HttpRequesterOperations.java

```diff
@@ -0,0 +1,165 @@
+                                                                             : new HttpRequesterRequestBuilder();
+
+        String resolvedHost = resolveIfNecessary(host, config.getHost(), muleEvent);
+        Integer resolvedPort = resolveIfNecessary(port, config.getPort(), muleEvent);
+        String resolvedBasePath = config.getBasePath().apply(muleEvent);
+        String resolvedPath = resolvedBuilder.replaceUriParams(buildPath(resolvedBasePath, path));
+
+        String resolvedUri = resolveUri(config.getScheme(), resolvedHost, resolvedPort, resolvedPath);
+        Boolean resolvedFollowRedirects = resolveIfNecessary(followRedirects, config.getFollowRedirects(), muleEvent);
+        HttpStreamingType resolvedStreamingMode = resolveIfNecessary(requestStreamingMode, config.getRequestStreamingMode(), muleEvent);
+        HttpSendBodyMode resolvedSendBody = resolveIfNecessary(sendBodyMode, config.getSendBodyMode(), muleEvent);
+        Boolean resolvedParseResponse = resolveIfNecessary(parseResponse, config.getParseResponse(), muleEvent);
+        Integer resolvedTimeout = resolveResponseTimeout(muleEvent, config, responseTimeout);
+        ResponseValidator resolvedValidator = responseValidator != null ? responseValidator
+                                                                        : new SuccessStatusCodeValidator("0..399");
+
+        HttpRequester requester = new HttpRequester.Builder()
+                .setUri(resolvedUri)
+                .setMethod(method)
+                .setFollowRedirects(resolvedFollowRedirects)
+                .setRequestStreamingMode(resolvedStreamingMode)
+                .setSendBodyMode(resolvedSendBody)
+                .setSource(source)
+                .setAuthentication(config.getAuthentication())
+                .setParseResponse(resolvedParseResponse)
+                .setResponseTimeout(resolvedTimeout)
+                .setResponseValidator(resolvedValidator)
+                .setConfig(config)
+                .build();
+
+        return requester.doRequest(muleEvent, client, resolvedBuilder, true);
+    }
+
+    private <T> T resolveIfNecessary(T value, Function<MuleEvent, T> function, MuleEvent event)
+    {
+        return value != null ? value : function.apply(event);
+    }
+
+    private String resolveUri(HttpConstants.Protocols scheme, String host, Integer port, String path)
+    {
+        // Encode spaces to generate a valid HTTP request.
+        String resolvedPath = HttpParser.encodeSpaces(path);
+
+        return String.format("%s://%s:%s%s", scheme, host, port, resolvedPath);
+    }
+
+    private int resolveResponseTimeout(MuleEvent muleEvent, HttpRequesterConfig config, Integer responseTimeout)
+    {
+        if (responseTimeout == null && config.getResponseTimeout() != null)
+        {
+            responseTimeout = config.getResponseTimeout().apply(muleEvent);
+        }
+
+        if (muleEvent.getMuleContext().getConfiguration().isDisableTimeouts())
```

Comment:
why do you use the context in the event instead of injecting the context into the operation?